packages/core/src/components/button/buttons.tsx

```diff
@@ -30,20 +32,63 @@ export interface IButtonProps extends IActionProps {
     loading?: boolean;
 }
 
-export class Button extends React.Component<React.HTMLProps<HTMLButtonElement> & IButtonProps, {}> {
+export interface IButtonState {
+    isActive: boolean;
+}
+
+export abstract class BaseButton<T> extends React.Component<React.HTMLProps<T> & IButtonProps, IButtonState> {
```

Comment:
Sweet, I'll leave moving the classes out, and also move all those helper functions into `AbstractButton` for another PR to not diverge *too* far from the original intent of this PR.